File: rust/tests/eval_seeds_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_eval_seeds_create_eval_seed_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .eval_seeds
        .create_eval_seed(
            &"project_id".to_string(),
            &CreateEvalSeedRequest {
                issue_id: "issueId".to_string(),
                issue_title: "issueTitle".to_string(),
                trace_ids: vec!["traceIds".to_string()],
                dataset_name: None,
                evaluator_name: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/projects/project_id/eval-seeds", None, 1)
        .await
        .unwrap();
}

File: ruby/test/wire/eval_seeds_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class EvalSeedsWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_eval_seeds_create_eval_seed_with_wiremock
    test_id = "eval_seeds.create_eval_seed.0"

    @client.eval_seeds.create_eval_seed(
      project_id: "project_id",
      issue_id: "issueId",
      issue_title: "issueTitle",
      trace_ids: ["traceIds"],
      request_options: {
        additional_headers: {
          "X-Test-Id" => "eval_seeds.create_eval_seed.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/eval-seeds",
      query_params: nil,
      expected: 1
    )
  end
end
